site/app/controllers/forum/ForumController.php

```diff
@@ -991,113 +1002,144 @@ public function splitThread() {
-            if (strlen($new_post_content) > ForumUtils::FORUM_CHAR_POST_LIMIT) {
-                $this->core->addErrorMessage("Posts cannot be over " . ForumUtils::FORUM_CHAR_POST_LIMIT . " characters long");
-                return null;
+        $content = filter_input(INPUT_POST, "thread_post_content", FILTER_UNSAFE_RAW);
+        if ($content === false || $content === "") {
+            return false;
+        }
+        if (strlen($content) > ForumUtils::FORUM_CHAR_POST_LIMIT) {
+            $this->core->addErrorMessage("Posts cannot be over " . ForumUtils::FORUM_CHAR_POST_LIMIT . " characters long");
+            return false;
+        }
+        $post->setContent($content);
+        if ($this->modifyAnonymous($post->getAuthor()->getId())) {
+            $post->setAnonymous(filter_input(INPUT_POST, "Anon", FILTER_UNSAFE_RAW) === "Anon");
+        }
+        $post->setRenderMarkdown(filter_input(INPUT_POST, "markdown_status", FILTER_VALIDATE_BOOL) !== false);
+
+        // TODO: Handle attachments & edit table
+        // As core gets legacy user model currently, which isn't a doctrine entity, we need to get the corresponding entity.
+        $edit_author = $this->core->getCourseEntityManager()->find(UserEntity::class, $this->core->getUser()->getId());
+        $post_edit = $post->saveNewVersion($edit_author);
+        $new_attachments = $this->saveAttachments(false, $post->getThread()->getId(), $post->getId(), 'file_input');
+        $em = $this->core->getCourseEntityManager();
+        if ($new_attachments !== false) {
+            foreach ($new_attachments as $attachment_name) {
+                $em->persist($post->addAttachment($attachment_name, $post_edit->getVersion()));
             }
-
-            $post_id = $_POST["edit_post_id"];
-            $original_post = $this->core->getQueries()->getPost($post_id);
-            $original_creator = !empty($original_post) ? $original_post['author_user_id'] : null;
-            $anon = (!empty($_POST["Anon"]) && $_POST["Anon"] == "Anon") ? 1 : 0;
-            $current_user = $this->core->getUser()->getId();
-            if (!$this->modifyAnonymous($original_creator)) {
-                $anon = $original_post["anonymous"] ? 1 : 0;
-            }
-
-            $markdown = !empty($_POST['markdown_status']);
-
-            $file_post = 'file_input';
-            $thread_id = $original_post["thread_id"];
-            $hasGoodAttachment = $this->checkGoodAttachment(false, $thread_id, $file_post);
-            if ($hasGoodAttachment[0] === -1) {
-                return null;
+        }
+        foreach (json_decode($_POST['deleted_attachments']) as $attachment_name) {
+            $post->deleteAttachment($attachment_name, $post_edit->getVersion());
+        }
+        if ($post->isChanged()) {
+            if (!is_null($initial_post)) {
+                $em->persist($initial_post);
             }
+            $em->persist($post_edit);
+        }
+        return true;
```

Comment:
Missing a flush?

Reply:
No need to flush after each persist; one at the very end will write all new persistent entities to database. Doctrine recommends 0-2 flushes per request.

Reply:
I see the flush happens after this call